OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1413 @@
+                        case `addDataPath` ⇒ dataPaths += (parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "") + parameter.last.asVar.value.toString.
+                            replaceAll(paramRegex, ""))
+                        case `addDataScheme` ⇒ dataSchemes += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                        case `addDataSSP` ⇒ dataSSPs += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                        case `addDataType` ⇒ dataTypes += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                    }
+                }
+            }
+            if (stm.isAssignment) {
+                if (stm.asAssignment.expr.asVirtualFunctionCall.name == registerReceiver) {
+                    if (stm.asAssignment.expr.asVirtualFunctionCall.receiver.asVar.definedBy.head > -1)
+                        registeredReceivers += ObjectType(statements(stm.asAssignment.expr.asVirtualFunctionCall.
+                            receiver.asVar.definedBy.head).asAssignment.expr.asGetField.objRef.asVar.value.toString.
+                            replaceAll(receiverRegex, "").replaceAll("\\.", "/"))
+                }
+            }
+        }
+    }
+}
+
+class implicitIntent(var caller: Method) {
+
+    var action: String = ""
+    var categories: ListBuffer[String] = ListBuffer.empty[String]
+    var iType: String = ""
+    var iPackage: String = ""
+    var iData: String = ""
+    var componentTypes: ListBuffer[String] = ListBuffer.empty[String]
+    var calledMethods: ListBuffer[String] = ListBuffer.empty[String]
+    var pc: Int = 0
+
+    def setAction(action: String): Unit = {
+        this.action = action
+    }
+
+    def addCategory(category: String): Unit = {
+        categories += category
+    }
+    def setData(uri: String): Unit = {
+        iData = uri
+    }
+
+    override def clone(): implicitIntent = {
+        val clone = new implicitIntent(caller)
+        clone.action = action
+        clone.categories = categories
+        clone.iType = iType
+        clone.iPackage = iPackage
+        clone.iData = iData
+        clone.componentTypes = componentTypes
+        clone.calledMethods = calledMethods
+        clone
+    }
+}
+
+/**
+ * Schedules the execution of an AndroidICCAnalysis.
+ * In order for the analysis to work the path to the projects AndroidManifest must be set via setManifest
+ */
+object eagerAndroidICCAnalysisScheduler extends BasicFPCFEagerAnalysisScheduler {
+    var manifestPath: String = _
+    def setManifest(manifestPath: String): Unit = {
+        this.manifestPath = manifestPath
+    }
+    override def requiredProjectInformation: ProjectInformationKeys = Seq(ComputeTACAIKey)
+
+    override def uses: Set[PropertyBounds] = Set(PropertyBounds.lb(Callers), PropertyBounds.lb(Callees))
```

Comment:
Use ProperyBounds.lbs(...) to create multiple at once.  However, this should almost certainly be ubs anyway (a lower bound is an overapproximation as an interim result, however, our call graphs usually produce underapproximations. I.e., they add edges as the computation progresses, not remove any).